### PR TITLE
fix(float)!: always anchor to corner of window including border

### LINKED ISSUE #15789: nvim_open_win(): behavior of south and east `anchor` is inconsistent across different `border` values when multigrid is not enabled
### Neovim version (nvim -v)

NVIM v0.6.0-dev+341-g05d685be52

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Arch Linux

### Terminal name/version

kitty 0.23.1

### $TERM environment variable

xterm-kitty

### Installation

AUR neovim-git

### How to reproduce the issue

1. Run `nvim --clean -O2`.
2. Type some text in the two buffers. Put cursor at somewhere near the center of the left window.
3. Run one of the following commands:
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': 'none' })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': 'single' })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': ['', '', '', '|', '', '', '', ''] })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': ['', '', '', '', '', '', '', '|'] })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': ['', '', '', '|', '', '', '', '|'] })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': ['', '-', '', '', '', '', '', ''] })`
    - `:call nvim_open_win(2, v:true, { 'relative': 'cursor', 'width': 8, 'height': 3, 'row': 0, 'col': 0, 'anchor': 'SE', 'border': ['', '', '', '', '', '-', '', ''] })`

### Expected behavior

Behavior of `anchor` is consistent across different `border` values.

### Actual behavior

Behavior of south and east `anchor` is inconsistent across different `border` values (maybe always treated as having no border).
The cursor is on the `*` before running the command which is also where the corner of the floating window should be anchored to.

No Border:
No Border

Full Border:
Full Border

Only Right Border:
Only Right Border

Only Left Border:
Only Left Border

Left and Right Border:
Left and Right Border

Only Top Border:
Only Top Border

Only Bottom Border:
Only Bottom Border

<details>
<summary>Northwest Anchor For Comparison</summary><br />

Full Border:
Northwest Anchor Full Border

No Border:
Northwest Anchor No Border

</details>

I'm not sure if the inconsistency should be fixed. If it is made consistent, some plugins may break and `vim.lsp.util.open_floating_preview()` may need changes, as they already depend on the inconsistent behavior.

If it should not be changed, I think the behavior of `anchor` should be better documented.

### PR DESCRIPTION
**Pull request by zeertzjq**
Fix #15789.

This make it unnecessary to call `get_border_size` in `make_floating_popup_options`, so testing `get_border_size` is hard now.

### CODE REVIEW

File: src/nvim/window.c [hunk 1]
```diff
@@ -853,11 +853,11 @@ void ui_ext_win_position(win_T *wp)
       bool east = c.anchor & kFloatAnchorEast;
       bool south = c.anchor & kFloatAnchorSouth;
 
-      int comp_row = (int)row - (south ? wp->w_height : 0);
-      int comp_col = (int)col - (east ? wp->w_width : 0);
+      int comp_row = (int)row - (south ? wp->w_height_outer - 1 : 0);
```
Comment (bfredl) [inline src/nvim/window.c:856]: why -1 here? It doesn't seem right.

Comment (zeertzjq) [reply src/nvim/window.c:856]: The difference in coordinates between top and botton sides is `w_height_outer - 1`.

Comment (bfredl) [reply src/nvim/window.c:856]: I'm pretty sure that difference is `w_height_outer`. That's what the variable is for.

Comment (zeertzjq) [reply src/nvim/window.c:856]: With the `- 1`:
Screenshot_20211001_154139
Screenshot_20211001_154157
Screenshot_20211001_154215
Screenshot_20211001_154230

Without the `- 1`:
Screenshot_20211001_154654
Screenshot_20211001_154708
Screenshot_20211001_154722
Screenshot_20211001_154736

Which one is intended?

Comment (bfredl) [reply src/nvim/window.c:856]: the latter. think of the anchor point as the upper-left corner of the marked cell, not the middle of it.

Comment (zeertzjq) [reply src/nvim/window.c:856]: OK, I change it.

**Pull request merged**